[docs] Update helm chart and discovery service references for kube app discovery (#31037)
gravitational/teleport: docs/pages/reference/helm-reference/teleport-kube-agent.mdx

| `Role`                | The release name, suffixed by `-psp`                       | Enforces security requirements for pods deployed by `teleport-kube-agent`.                                                                                                                                                                                                                              | `podSecurityPolicy.enabled` is `true` and the Kubernetes cluster version is < 1.23.                               |
| `RoleBinding`         | The release name, suffixed by `-psp`                       | Enforces security requirements for pods deployed by `teleport-kube-agent`.                                                                                                                                                                                                                              | `podSecurityPolicy.enabled` is `true` and the Kubernetes cluster version is < 1.23.                               |

> ## Values reference

Reviewer: Under this new H2, I would bump all intended subsections up a level, i.e., so H2s become H3s. We can then adjust `tocDepth` in the frontmatter to `3`.

Otherwise, this looks a bit odd in the table of contents.

Author: I'll leave it to a separate PR (together with new H2) to not mix content. This change is not directly related to the content of this PR.

  Once `databaseResources` is set, you can dynamically register database with `tsh` by following [this guide](../../database-access/guides/dynamic-registration.mdx).
</Admonition>

## kubernetesDiscovery

| Type   | Default value | Required?                                                                                                                             |
|--------|---------------|---------------------------------------------------------------------------------------------------------------------------------------|

> | `list` | `See below`   | When the `discovery` chart role is used, `kubernetesDiscovery` is required. |

Reviewer: ```suggestion
| `list` | Read access to all applications in all cluster namespaces.   | When the `discovery` chart role is used, `kubernetesDiscovery` is required. |
```
I would change `See below` to a description of the default access permissions. Otherwise, (a) `See below` seems like a default value itself, which is a bit unexpected and (b) it's not very explicit where a reader should look.

Feel free to change my suggested replacement to something more accurate!

Default value:

  ```yaml
  kubernetesDiscovery:
  - types: ["app"]
    namespaces: ["*"]
    labels:
      "*": "*"
  ```

`kubernetesDiscovery` is a YAML list object detailing the filters for the services of the Kubernetes cluster that should be discovered
 and exposed as apps in Teleport. By default the Discovery Service will try to find all apps that are serving HTTP/HTTPS, but you can manually

>  make it expose TCP apps as well, using Kubernetes annotations.

Reviewer: I would link to the reference on using Kubernetes annotations.

  Once `databaseResources` is set, you can dynamically register database with `tsh` by following [this guide](../../database-access/guides/dynamic-registration.mdx).
</Admonition>

## kubernetesDiscovery

| Type   | Default value | Required?                                                                                                                             |
|--------|---------------|---------------------------------------------------------------------------------------------------------------------------------------|
| `list` | `See below`   | When the `discovery` chart role is used, `kubernetesDiscovery` is required. |

Default value:

  ```yaml
  kubernetesDiscovery:
  - types: ["app"]
    namespaces: ["*"]
    labels:
      "*": "*"
  ```

`kubernetesDiscovery` is a YAML list object detailing the filters for the services of the Kubernetes cluster that should be discovered
 and exposed as apps in Teleport. By default the Discovery Service will try to find all apps that are serving HTTP/HTTPS, but you can manually
 make it expose TCP apps as well, using Kubernetes annotations.

You can specify multiple filters, each of which can include these parameters:

> - `types` is a list of types of services to be exposed in Teleport, currently only "app" value is supported, with "db" value planned.

Reviewer: ```suggestion
- `types` is a list of types of services to be exposed in Teleport. Currently only `app` is supported.
```
Grammar tweak. Also removing promises, which I don't think will help a reader looking for current documentation.

>  Default value is "app"

Reviewer: ```suggestion
```
Since this is the only possible value.

> - `namespaces` is a list of namespaces of the Kubernetes cluster to search in. Default value is `*`, which denotes all namespaces.

Reviewer: ```suggestion
- `namespaces` is a list of namespaces of the Kubernetes cluster to search in. The default value is `*`, which denotes all namespaces.
```

> - `labels` is a list of Kubernetes labels based on which to filter out services. Labels  Default value is `*`: `*` that will match any labels.

Reviewer: ```suggestion
- `labels` is a list of Kubernetes labels based on which to filter out services. The default value is `*`: `*` that will match any labels.
```
